pkg/recipes/driver/gitconfig.go

```diff
@@ -88,19 +101,30 @@ func addSecretsToGitConfig(secrets v20231001preview.SecretStoresClientListSecret
 	return nil
 }
 
-// Unset the git credentials information from .gitconfig by running
-// git config --global --unset url<template_path_domain_with_credentails>.insteadOf
-func unsetSecretsFromGitConfig(secrets v20231001preview.SecretStoresClientListSecretsResponse, templatePath string) error {
-	urlConfigKey, _, err := getURLConfigKeyValue(secrets, templatePath)
+// setGitConfigForDir sets a conditional include directive in the global Git configuration file.
+// This function modifies the global Git configuration to include a specific Git configuration file
+// when the repository is located in the given working directory. The `includeIf` directive is used
+// to conditionally include the configuration file located at "<workingDirectory>/.git/config".
+func setGitConfigForDir(workingDirectory string) error {
+	cmd := exec.Command("git", "config", "--global", fmt.Sprintf("includeIf.gitdir:%s/.path", workingDirectory), workingDirectory+"/.git/config")
+	_, err := cmd.Output()
 	if err != nil {
-		return err
+		return errors.New("failed to add conditional include directive")
 	}
+	return nil
+}
 
-	cmd := exec.Command("git", "config", "--global", "--unset", urlConfigKey)
-	_, err = cmd.Output()
+// unsetGitConfigForDir removes a conditional include directive from the global Git configuration.
+// This function modifies the global Git configuration to remove a previously set `includeIf` directive
+// for a given working directory.
+func unsetGitConfigForDir(workingDirectory string, secrets v20231001preview.SecretStoresClientListSecretsResponse, templatePath string) error {
+	if !strings.HasPrefix(templatePath, "git::") || reflect.DeepEqual(secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
+		return nil
+	}
+	cmd := exec.Command("git", "config", "--global", "--unset", fmt.Sprintf("includeIf.gitdir:%s/.path", workingDirectory))
+	_, err := cmd.Output()
 	if err != nil {
-		return errors.New("failed to unset git config")
+		return errors.New("failed to unset conditional include directive")
 	}
-
```

Comment:
Is this being auto formatted by some setting in vs code?

Reply:
no, i wonder if this factoring happens on save or is it manual for others?

Reply:
haven't seen this happen on vscode for me, so maybe a local setting that automatically refactors it this way